src/frequenz/client/microgrid/_component_error.py

```diff
@@ -0,0 +1,198 @@
+    """Charge current has exceeded component bounds."""
+
+    EXCEEDED_SOP_DISCHARGE = battery.ErrorCode.ERROR_CODE_EXCEEDED_SOP_DISCHARGE
+    """Discharge current has exceeded component bounds."""
+
+    SYSTEM_IMBALANCE = battery.ErrorCode.ERROR_CODE_SYSTEM_IMBALANCE
+    """The battery blocks are not balanced with respect to each other."""
+
+    LOW_SOH = battery.ErrorCode.ERROR_CODE_LOW_SOH
+    """The State of health is low."""
+
+    BLOCK_ERROR = battery.ErrorCode.ERROR_CODE_BLOCK_ERROR
+    """One or more battery blocks have failed."""
+
+    CONTROLLER_ERROR = battery.ErrorCode.ERROR_CODE_CONTROLLER_ERROR
+    """The battery controller has failed."""
+
+    RELAY_ERROR = battery.ErrorCode.ERROR_CODE_RELAY_ERROR
+    """The battery's DC relays have failed."""
+
+    RELAY_CYCLE_LIMIT_REACHED = battery.ErrorCode.ERROR_CODE_RELAY_CYCLE_LIMIT_REACHED
+    """The battery's DC relays have reached the cycles limit in its lifetime specifications."""
+
+    FUSE_ERROR = battery.ErrorCode.ERROR_CODE_FUSE_ERROR
+    """The battery's fuse has failed."""
+
+    EXTERNAL_POWER_SWITCH_ERROR = (
+        battery.ErrorCode.ERROR_CODE_EXTERNAL_POWER_SWITCH_ERROR
+    )
+    """The eternal power switch has failed."""
+
+    PRECHARGE_ERROR = battery.ErrorCode.ERROR_CODE_PRECHARGE_ERROR
+    """The precharge operation has failed."""
+
+    SYSTEM_PLAUSIBILITY_ERROR = battery.ErrorCode.ERROR_CODE_SYSTEM_PLAUSIBILITY_ERROR
+    """System plausibility checks have failed."""
+
+    SYSTEM_UNDERVOLTAGE_SHUTDOWN = (
+        battery.ErrorCode.ERROR_CODE_SYSTEM_UNDERVOLTAGE_SHUTDOWN
+    )
+    """System shut down due to extremely low voltage."""
+
+    CALIBRATION_NEEDED = battery.ErrorCode.ERROR_CODE_CALIBRATION_NEEDED
+    """The battery requires a calibration to reset its measurements."""
+
+    @classmethod
+    def from_pb(cls, code: battery.ErrorCode) -> Self:
+        """Convert a protobuf error code value to this enum.
+
+        Args:
+            code: The protobuf error code to convert.
+
+        Returns:
+            The enum value corresponding to the protobuf message.
+        """
+        try:
+            return cls(code)
+        except ValueError:
+            return cls(cls.UNSPECIFIED)
+
+
+@dataclass(frozen=True, kw_only=True)
+class BatteryError:
+    """A battery error."""
+
+    code: BatteryErrorCode = BatteryErrorCode.UNSPECIFIED
+    """The error code."""
+
+    level: ErrorLevel = ErrorLevel.UNSPECIFIED
+    """The error level."""
+
+    message: str = ""
+    """The error message."""
+
+    @classmethod
+    def from_pb(cls, raw: battery.Error) -> Self:
+        """Create a new instance using a protobuf message to get the values.
+
+        Args:
+            raw: The protobuf message to get the values from.
+
+        Returns:
+            The new instance with the values from the protobuf message.
+        """
+        return cls(
+            code=BatteryErrorCode.from_pb(raw.code),
+            level=ErrorLevel(raw.level),
```

Comment:
I think we shouldn't assume there will only be `WARN` and `CRITICAL` levels, because we don't know what kind of new component types encounter and what kind of error levels they have.  Much safer to have a `from_pb` for `ErrorLevel` too, like the other enum types.